File: android/app/src/main/java/com/example/demo/ui/components/SubscriptionCard.kt
package com.example.demo.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Check
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.demo.data.SubscriptionPlan

@Composable
fun SubscriptionCard(
    plan: SubscriptionPlan,
    modifier: Modifier = Modifier,
    onSelectPlan: () -> Unit = {},
) {
    Card(
        modifier =
            modifier
                .fillMaxWidth()
                .height(350.dp),
        shape = RoundedCornerShape(16.dp),
        border =
            if (plan.isHighlighted) {
                BorderStroke(2.dp, MaterialTheme.colorScheme.primary)
            } else {
                BorderStroke(1.dp, Color.Gray.copy(alpha = 0.3f))
            },
        elevation =
            CardDefaults.cardElevation(
                defaultElevation = if (plan.isHighlighted) 8.dp else 4.dp,
            ),
    ) {
        Box(modifier = Modifier.fillMaxSize()) {
            // Most Popular Badge
            if (plan.isHighlighted) {
                Box(
                    modifier =
                        Modifier
                            .align(Alignment.TopCenter)
                            .offset(y = (-12).dp)
                            .background(
                                MaterialTheme.colorScheme.primary,
                                RoundedCornerShape(12.dp),
                            )
                            .padding(horizontal = 12.dp, vertical = 4.dp),
                ) {
                    Text(
                        text = "Most Popular",
                        color = MaterialTheme.colorScheme.onPrimary,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Medium,
                    )
                }
            }

            Column(
                modifier =
                    Modifier
                        .fillMaxSize()
                        .padding(16.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
            ) {
                // Plan Name
                Text(
                    text = plan.name,
                    style = MaterialTheme.typography.headlineSmall,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    modifier = Modifier.padding(top = if (plan.isHighlighted) 8.dp else 0.dp),
                )

                // Price
                Text(
                    text = plan.price,
                    style = MaterialTheme.typography.headlineLarge,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onSurface,
                    modifier = Modifier.padding(vertical = 8.dp),
                )

                // Features List
                Column(
                    modifier =
                        Modifier
                            .weight(1f)
                            .padding(vertical = 8.dp),
                    verticalArrangement = Arrangement.spacedBy(8.dp),
                ) {
                    plan.features.forEach { feature ->
                        Row(
                            verticalAlignment = Alignment.CenterVertically,
                            modifier = Modifier.fillMaxWidth(),
                        ) {
                            Icon(
                                imageVector = Icons.Default.Check,
                                contentDescription = null,
                                tint = MaterialTheme.colorScheme.primary,
                                modifier = Modifier.size(16.dp),
                            )
                            Spacer(modifier = Modifier.width(8.dp))
                            Text(
                                text = feature,
                                style = MaterialTheme.typography.bodyMedium,
                                color = MaterialTheme.colorScheme.onSurfaceVariant,
                            )
                        }
                    }
                }

                // Action Button
                Button(
                    onClick = onSelectPlan,
                    modifier =
                        Modifier
                            .fillMaxWidth()
                            .padding(top = 8.dp),
                    colors =
                        if (plan.isHighlighted) {
                            ButtonDefaults.buttonColors()
                        } else {
                            ButtonDefaults.outlinedButtonColors()
                        },
                ) {
                    Text(
                        text = if (plan.isHighlighted) "Get Started" else "Select Plan",
                        fontWeight = FontWeight.Medium,
                    )
                }
            }
        }
    }
}

File: ios/ConfidenceDemoApp/UI/Components/SubscriptionCard.swift
import SwiftUI

struct SubscriptionCard: View {
    let plan: SubscriptionPlan
    let onSelectPlan: () -> Void

    @Environment(\.materialTheme) var theme

    var body: some View {
        ZStack {
            RoundedRectangle(cornerRadius: 16)
                .fill(theme.surface)
                .overlay(
                    RoundedRectangle(cornerRadius: 16)
                        .stroke(
                            plan.isHighlighted ? theme.primary : Color.gray.opacity(0.3),
                            lineWidth: plan.isHighlighted ? 2 : 1
                        )
                )
                .shadow(
                    color: Color.black.opacity(0.1),
                    radius: plan.isHighlighted ? 8 : 4,
                    x: 0,
                    y: 2
                )

            VStack(spacing: 0) {
                // Most Popular Badge
                if plan.isHighlighted {
                    HStack {
                        Spacer()
                        Text("Most Popular")
                            .font(.caption)
                            .fontWeight(.medium)
                            .foregroundColor(.white)
                            .padding(.horizontal, 12)
                            .padding(.vertical, 4)
                            .background(theme.primary)
                            .clipShape(RoundedRectangle(cornerRadius: 12))
                            .offset(y: -12)
                        Spacer()
                    }
                    .zIndex(1)
                }

                VStack(spacing: 16) {
                    VStack(spacing: 8) {
                        // Plan Name
                        Text(plan.name)
                            .font(.title2)
                            .fontWeight(.bold)
                            .foregroundColor(theme.onSurfaceVariant)
                            .padding(.top, plan.isHighlighted ? 0 : 16)

                        // Price
                        Text(plan.price)
                            .font(.title)
                            .fontWeight(.bold)
                            .foregroundColor(theme.onSurface)
                    }

                    // Features List
                    VStack(alignment: .leading, spacing: 8) {
                        ForEach(plan.features, id: \.self) { feature in
                            HStack(alignment: .top, spacing: 8) {
                                Image(systemName: "checkmark")
                                    .foregroundColor(theme.primary)
                                    .font(.caption)
                                    .frame(width: 16, height: 16)

                                Text(feature)
                                    .font(.body)
                                    .foregroundColor(theme.onSurfaceVariant)
                                    .multilineTextAlignment(.leading)

                                Spacer(minLength: 0)
                            }
                        }
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)

                    Spacer()

                    // Action Button
                    Button(action: onSelectPlan) {
                        Text(plan.isHighlighted ? "Get Started" : "Select Plan")
                            .fontWeight(.medium)
                            .foregroundColor(plan.isHighlighted ? .white : theme.primary)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, 12)
                            .background(
                                plan.isHighlighted ? theme.primary : Color.clear
                            )
                            .overlay(
                                RoundedRectangle(cornerRadius: 8)
                                    .stroke(theme.primary, lineWidth: plan.isHighlighted ? 0 : 1)
                            )
                            .clipShape(RoundedRectangle(cornerRadius: 8))
                    }
                }
                .padding(16)
            }
        }
        .frame(height: 350)
    }
}